### ISSUE TITLE
Add `jax.nn.normalize`

### ISSUE THREAD

**Issue by carlosgmartin** [google/jax#20556]
[`jax.nn`](https://jax.readthedocs.io/en/latest/jax.nn.html#other-functions) contains the function [`jax.nn.standardize`](https://jax.readthedocs.io/en/latest/_autosummary/jax.nn.standardize.html), which does [standardization](https://en.wikipedia.org/wiki/Feature_scaling#Standardization_(Z-score_Normalization)).

Another common operation in data processing is [min-max normalization](https://en.wikipedia.org/wiki/Feature_scaling#Rescaling_(min-max_normalization)) to some interval, usually [0, 1] or [-1, 1].

**Feature request:** Add a function `jax.nn.normalize` that does this.

Possible implementation:

```python3
def normalize(a, axis=None, where=None, lb=0, ub=1):
    """Applies a min-max normalization of values to the interval [lb, ub]."""
    a_max = a.max(axis=axis, where=where, keepdims=True, initial=-jnp.inf)
    a_min = a.min(axis=axis, where=where, keepdims=True, initial=+jnp.inf)
    a_ptp = a_max - a_min
    a_ptp = jnp.where(a_ptp == 0, 1, a_ptp)  # avoid division by zero
    a_norm = (a - a_min) / a_ptp
    return lb + (ub - lb) * a_norm
```

I can submit a PR for this.

**Comment by jakevdp** [comment]
`jax.nn.normalize` already exists: it's a deprecated alias to `jax.nn.standardize`. Before we add another function of the same name, we'll have to finalize the deprecation, and then it would probably be good to wait several releases before introducing a different `normalize` function.

**Comment by carlosgmartin** [comment]
I see. How long would that take?

Would you prefer to wait until then, or to use a different name, such as `minmax_normalize`?

**Comment by jakevdp** [comment]
I'm not entirely convinced of the need for this function in `jax.nn`.

**Comment by carlosgmartin** [comment]
@jakevdp I think [min-max normalization](https://en.wikipedia.org/wiki/Feature_scaling#Rescaling_(min-max_normalization)) is one of the most common procedures in machine learning and data science, perhaps even more common than standardization. See scikit-learn's [minmax_scale](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.minmax_scale.html) and [MinMaxScaler](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MinMaxScaler.html). A [GitHub code search](https://github.com/search?q=jax+%28minmaxnormalize+OR+minmax_normalize+OR+min_max_normalize%29&type=code) confirms its widespread use, even for JAX specifically.

If you're open to this, I'll create a PR adding a function `jax.nn.min_max_normalize` along the lines described above.
